mgnify_pipelines_toolkit/analysis/assembly/go_utils.py

```diff
@@ -0,0 +1,57 @@
+import os
+import re
+
+
+def count_and_assign_go_annotations(go2protein_count, go_annotations, num_of_proteins):
+    for go_id in go_annotations:
+        count = go2protein_count.setdefault(go_id, 0)
+        count += 1 * num_of_proteins
+        go2protein_count[go_id] = count
+
+
+def parse_ips_file(ips_file):
+
+    go2protein_count = {}
+    num_of_proteins_with_go = 0
+    total_num_of_proteins = 0
+    if os.path.exists(ips_file):
```

Comment:
I would invert this, `if no os.path.exists(...)` then return the go2protein_count (whch is going to be empty).

That removes the identation making reading the thing easier IMO